src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.NativeAot.cs

```diff
@@ -1632,4 +1645,217 @@ private static unsafe IntPtr ObjectToComWeakRef(object target, out long wrapperI
+    {
+        private const int DefaultSize = 7;
+
+        private Entry?[] _buckets = new Entry[DefaultSize];
+        private int _numEntries;
+        private readonly Lock _lock = new Lock(useTrivialWaits: true);
+
+        public void Add(GCHandle handle)
+        {
+            using (_lock.EnterScope())
+            {
+                int bucket = GetBucket(handle);
+                Entry? prev = null;
+                Entry? entry = _buckets[bucket];
+                while (entry != null)
+                {
+                    // Handle already exists, nothing to add.
+                    if (handle.Equals(entry.m_value))
+                    {
+                        return;
+                    }
+
+                    prev = entry;
+                    entry = entry.m_next;
+                }
+
+                Entry newEntry = new Entry()
+                {
+                    m_value = handle
+                };
+
+                if (prev == null)
+                {
+                    _buckets[bucket] = newEntry;
+                }
+                else
+                {
+                    prev.m_next = newEntry;
+                }
+
+                // _numEntries is only maintained for the purposes of deciding whether to
+                // expand the bucket and is not used during iteration to handle the
+                // scenario where element is in bucket but _numEntries hasn't been incremented
+                // yet.
+                _numEntries++;
+                if (_numEntries > (_buckets.Length * 2))
+                {
+                    ExpandBuckets();
+                }
+            }
+        }
+
+        private void ExpandBuckets()
+        {
+            try
+            {
+                int newNumBuckets = _buckets.Length * 2 + 1;
+                Entry?[] newBuckets = new Entry[newNumBuckets];
+                for (int i = 0; i < _buckets.Length; i++)
+                {
+                    Entry? entry = _buckets[i];
+                    while (entry != null)
+                    {
+                        Entry? nextEntry = entry.m_next;
+
+                        int bucket = GetBucket(entry.m_value, newNumBuckets);
+
+                        // We are allocating new entries for the bucket to ensure that
+                        // if there is an enumeration already in progress, we don't
+                        // modify what it observes by changing next in existing instances.
+                        Entry newEntry = new Entry()
+                        {
+                            m_value = entry.m_value,
+                            m_next = newBuckets[bucket],
+                        };
+                        newBuckets[bucket] = newEntry;
+
+                        entry = nextEntry;
+                    }
+                }
+                _buckets = newBuckets;
+            }
+            catch (OutOfMemoryException)
+            {
+            }
```

Comment:
```suggestion
```
I do not think you need to bother with catching `OutOfMemoryException`. It won't help the app to survive much longer.